Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -286,19 +289,26 @@ IndexInfo::globalSpectrumIndicesFromDetectorIndices(
           m_spectrumNumberTranslator->partitionOf(GlobalSpectrumIndex(i)));
       const auto spectrumDefinition =
           spectrumDefinitions[rank][currentIndex[rank]++];
-      if (spectrumDefinition >= 0) {
-        const auto detectorIndex = static_cast<size_t>(spectrumDefinition);
+      if (spectrumDefinition.first >= 0) {
+        const auto detectorIndex =
+            static_cast<size_t>(spectrumDefinition.first);
+        const auto timeIndex = static_cast<size_t>(spectrumDefinition.second);
         if (detectorMap.size() > detectorIndex &&
-            detectorMap[detectorIndex] != 0) {
-          if (detectorMap[detectorIndex] > 1)
-            throw std::runtime_error(
-                "Multiple spectra correspond to the same detector");
-          // Increment flag to catch two spectra mapping to same detector.
-          ++detectorMap[detectorIndex];
+            detectorMap[detectorIndex].first != 0) {
           spectrumIndices.push_back(i);
+          if (detectorMap[detectorIndex].second.size() <= timeIndex) {
+            detectorMap[detectorIndex].second.resize(timeIndex + 1, {0});
+            detectorMap[detectorIndex].second[timeIndex] = 1;
+          }
+          if (detectorMap[detectorIndex].first == 1) {
+            ++detectorMap[detectorIndex].first;
+          }
+          if (detectorMap[detectorIndex].second[timeIndex] != 0) {
+            ++detectorMap[detectorIndex].second[timeIndex];
+          }
```

Comment:
I had trouble understanding what is going on here --- suddenly the meaning/purpose of `detectorMap` is changing (from indicating which detectors are requested) to indicating which detectors have a matching mapping (value set to 2 or higher). Can you explain this in a comment?

Reply:
I meant to add a comment in the code to explain this, because it took me quite a long time to understand.

Reply:
Good point.